### PR TITLE
Update: Add `CallExpression` option for `indent` (fixes #5946)

### LINKED ISSUE #5946: Proposed Rule: align function arguments
**When does this rule warn? Please describe and show example code:**

The rule warns when arguments in a multi-line function call are not aligned.

This code will warn:

```js
var result = myFunc('some long string that will cause someone to want to wrap',
  bar
  baz);
```

The following will not warn:

```js
var result = myFunc(somethingLongThatMightCauseSomeoneToWantToWrap,
                    bar
                    baz);
```

If an argument is itself multiline (for example, an inline function) then the rule does not fire for the next argument. This means this code will not warn even though the second argument (`100`) is not aligned with the first (the function):

```js
setTimeout(function () {
  // do some stuff
}, 100);
```

**Is this rule preventing an error or is it stylistic?**

Stylistic

**Why is this rule a candidate for inclusion instead of creating a custom rule?**

It is a widespread practice that some developers will want to enable linting for. Some developers may already expect this to be possible.

**Are you willing to create the rule yourself?**

Sure. I have a first implementation here: https://github.com/Trott/io.js/blob/align-multiline-function-call/tools/eslint-rules/align-function-arguments.js

### PR DESCRIPTION
**Pull request by not-an-aardvark**
**What is the purpose of this pull request? (put an "X" next to item)**

[ ] Documentation update
[ ] Bug fix ([template](https://github.com/eslint/eslint/blob/master/templates/bug-report.md))
[ ] New rule ([template](https://github.com/eslint/eslint/blob/master/templates/rule-proposal.md))
[x] Changes an existing rule ([template](https://github.com/eslint/eslint/blob/master/templates/rule-change-proposal.md))
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

**What rule do you want to change?**

[`indent`](http://eslint.org/docs/rules/indent)

**Does this change cause the rule to produce more or fewer warnings?**

The rule produces more warnings by opt-in.

**How will the change be implemented? (New option, new default behavior, etc.)?**

A new `CallExpression` option is added to the `indent` rule, to specify indentation for the arguments of `CallExpression` nodes. (Also see: #5946)

**Please provide some example code that this change will affect:**

Examples of **incorrect** code for this rule with the `2, { "CallExpression": {"arguments": 1} }` option:

```js
/*eslint indent: ["error", 2, { "CallExpression": {"arguments": 1} }]*/

foo(bar,
    baz,
      qux
);
```

Examples of **correct** code for this rule with the `2, { "CallExpression": {"arguments": 1} }` option:

```js
/*eslint indent: ["error", 2, { "CallExpression": {"arguments": 1} }]*/

foo(bar,
  baz,
  qux
);
```

Examples of **incorrect** code for this rule with the `2, { "CallExpression": {"arguments": "first"} }` option:

```js
/*eslint indent: ["error", 2, {"CallExpression": {"arguments": "first"}}]*/

foo(bar, baz,
  baz, boop, beep);
```

Examples of **correct** code for this rule with the `2, { "CallExpression": {"arguments": "first"} }` option:

```js
/*eslint indent: ["error", 2, {"CallExpression": {"arguments": "first"}}]*/

foo(bar, baz,
    baz, boop, beep);
```

**What does the rule currently do for this code?**

It raises an error because the schema is invalid. Without the `CallExpression` option, the indentation of CallExpression arguments is currently not linted at all.

**What will the rule do after it's changed?**

It will check the indentation of `CallExpression` arguments (see above).

**Please check each item to ensure your pull request is ready:**

- [x] I've read the [pull request guide](http://eslint.org/docs/developer-guide/contributing/pull-requests)
- [x] I've included tests for my change
- [x] I've updated documentation for my change (if appropriate)

**What changes did you make? (Give an overview)**

This adds a `CallExpression` option to the `indent` rule. The option is an object with one key: `arguments`.

If the value of `arguments` is a number, the rule enforces that indentation level for `CallExpression` arguments (similar to all the other `indent` options).

If the value of `arguments` is the string `"first"`, all arguments in the CallExpression must be aligned with the first argument of the expression (similar to `{FunctionDeclaration: {parameters: "first"}}` and `{FunctionExpression: {parameters: "first"}}`).

**Is there anything you'd like reviewers to focus on?**

I wasn't sure about how to design the schema; at the moment it's `{CallExpression: {arguments: 1}}`, but it could also just be `{CallExpression: 1}` since `arguments` is the only key of that object. I went with this schema because it makes it clearer that the option deals with the *arguments* of the CallExpression, rather than the expression itself.

### CODE REVIEW

File: tests/lib/rules/indent.js [hunk 1]
```diff
@@ -1504,7 +1504,7 @@ ruleTester.run("indent", rule, {
             "{\n" +
             "      bar();\n" +
             "}",
-            options: [2, {FunctionDeclaration: {body: 3}}] // FIXME: what is the default for `parameters`?
+            options: [2, {FunctionDeclaration: {body: 3}}]
```
Comment (not-an-aardvark) [inline tests/lib/rules/indent.js]: (This removal isn't actually related to this PR; I forgot to remove the comments in #7076.)

**Comment by kaicataldo** [comment]
I know we're still evaluating, but looks like there's a merge conflict

**Comment by not-an-aardvark** [comment]
I'll champion this.

**Comment by vitorbal** [comment]
I'll support this. The current schema is nice since it has some symmetry with the schemas of `{FunctionDeclaration: {parameters: "first"}}` and `{FunctionExpression: {parameters: "first"}}`, so :+1: from me on that too.

File: lib/rules/indent.js [hunk 2]
```diff
@@ -113,6 +113,22 @@ module.exports = {
...
+                        type: "object",
+                        properties: {
+                            parameters: {
+                                oneOf: [
+                                    {
+                                        type: "integer",
+                                        minimum: 0
+                                    },
+                                    {
+                                        enum: ["first"]
```
Comment (not-an-aardvark) [inline lib/rules/indent.js]: I see your point about this, but if we're going to explicitly add `"off"` values to options that are disabled by default, I think it should be part of a separate PR, since at the moment there are a lot of other `indent` options that don't have `"off"`.
